packages/block-editor/src/components/block-tools/selected-block-popover.js

```diff
@@ -31,7 +32,7 @@ function selector( select ) {
 		__experimentalIsBlockInterfaceHidden: isBlockInterfaceHidden,
 		getSettings,
 		getLastMultiSelectedBlockClientId,
-	} = select( blockEditorStore );
+	} = unlock( select( blockEditorStore ) );
```

Comment:
So this is the bit that triggers an error. I haven't tested the actions yet.

The error is `Uncaught TypeError: isBlockInterfaceHidden is not a function`.

When debugging I can see that `privateSelectors` [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/redux-store/index.js#L192) doesn't include the `__experimentalIsBlockInterfaceHidden` property.

It may be a race condition? I can see that:
1. Store is instantiated, [`selectors` is locked and proxied while empty](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/redux-store/index.js#L185-L202).
2. [`registerPrivateSelectors` is called](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/redux-store/index.js#L267-L269), and after the Object.assign, `privateSelectors` DOES include the `__experimentalIsBlockInterfaceHidden` property.
3. When selecting a group block in the UI, the proxy `get` function is correctly called, [but `privateSelectors is empty](https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/redux-store/index.js#L192), like in step 1.

Reply:
@talldan It's being thrown off by `withRegistryProvider` which isn't supported at the moment. Conceptually, registering the same store in a children registry should expose the same experimental actions and selectors as the parent registry. The fix would be to add this proxying to one or more of:

* `packages/data/src/redux-store/index.js`
* `packages/data/src/registry.js`
* `packages/block-editor/src/components/provider/with-registry-provider.js`

I tried to prepare a PR today but didn't get anywhere yet.

Reply:
In theory, something like this would solve it:

```js
lock(
    subRegistry.dispatch( childrenStore),
    unlock( parentRegistry.dispatch( parentStore ) )
)
```

An even better fix would adjust the `Proxy` implementation to also look in the parent registry.

Reply:
I found one more issue:

```
export const store = createReduxStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );

// Ideally we'd use register instead of register stores.
registerStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );

unlock( store ).registerPrivateActions( {
	test: () => {},
} );
```

In this setting, private actions are registered on the store that gets created, but not on the one that gets registered. One more reason to get rid of `registerStore` eventually. 

The following change in registry.js solves it:

```js
	function registerStore( storeName, options ) {
		if ( ! options.reducer ) {
			throw new TypeError( 'Must specify store reducer' );
		}

		const descriptor = createReduxStore( storeName, options );
		const store = descriptor.instantiate( registry );
		registerStoreInstance( storeName, store );
		lock( store.store, unlock( descriptor ) );
		return store.store;
	}
```

And then the private action registration goes like this:

```js
export const store = createReduxStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );

const registeredStore = registerStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );

unlock( registeredStore ).registerPrivateActions( {
	test: () => {},
} );
```

Reply:
I addressed the issue as a part of https://github.com/WordPress/gutenberg/pull/47319/. I'll extract it into a separate PR to make the discussion easier.

Reply:
There you go: https://github.com/WordPress/gutenberg/pull/47421

Reply:
If this lands: https://github.com/WordPress/gutenberg/pull/47447, we can merge independently of https://github.com/WordPress/gutenberg/pull/47421

Reply:
Both PRs landed, @talldan this issue should be fixed now